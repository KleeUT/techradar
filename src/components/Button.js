import React from 'react';
import styled from 'styled-components';
export default ({children, onClick}) => {
    const Button = styled.button`
        padding:0.5em;
        background-color:#404;
        color:#fff;
        border:0;
        margin:0.5em;
    `

    return <Button onClick={onClick}>{children}</Button>
}