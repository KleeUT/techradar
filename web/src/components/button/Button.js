import React from "react";
import styled from "styled-components";
import propTypes from "prop-types";
import { ActiveDutyGreen, NonCombatantGreen } from '../../util/Colors'
const Button = styled.button`
        padding:0.75rem;
        background-color: ${props => props.canClick ? ActiveDutyGreen : NonCombatantGreen };
        color: ${props => props.canClick ? '#fff' : '#999' };
        border:0;
        margin:0.5rem;
        border-radius: 3px;
        font-family: 'Roboto', sans-serif;
        font-size:16px;
        min-width:5rem;
        cursor: ${props => props.canClick ? 'pointer' : 'not-allowed'};
    `;
const Component = ({ children, onClick, canClick = true }) => {
  return (
    <Button onClick={onClick} canClick={canClick} disabled={ !canClick }>
      {children}
    </Button>
  );
};

Component.propTypes = {
  children: propTypes.string,
  onClick: propTypes.func,
  canClick: propTypes.bool
};

export default Component;
