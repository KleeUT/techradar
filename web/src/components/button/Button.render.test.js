import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import renderer from 'react-test-renderer';
// import renderer from 'react-test-renderer';

describe('Button: ', () => {
  it(' renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button onClick={() => console.log('something')}>With Text</Button>, div);
  });
 
  it(' passes snapshot tests', () => {
    const component = renderer.create(
      <Button>Snap Shot Tests</Button>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})