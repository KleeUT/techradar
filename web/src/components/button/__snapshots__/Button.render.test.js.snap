// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button:   passes snapshot tests 1`] = `
<button
  className="sc-bdVaJa iCSLLn"
  disabled={false}
  onClick={undefined}
>
  Snap Shot Tests
</button>
`;
